scheduler/util.go

```diff
@@ -255,7 +255,7 @@ func readyNodesInDCs(state State, dcs []string) ([]*structs.Node, map[string]int
 		if node.Status != structs.NodeStatusReady {
 			continue
 		}
-		if node.Drain {
+		if node.DrainStrategy != nil {
```

Comment:
Similar to the plan applier logic, we actually don't need to check this at all anymore. Since enforce `SchedulingEligibilty==Ineligible when DrainStategy!=nil` in `Node.Canonicalize()` we can skip this check as we check eligibility below.

Reply:
I updated `Canonicalize` to enforce that